Kconfig

```diff
@@ -9,6 +9,7 @@ mainmenu "RIOT Configuration"
 # For now, get used modules as macros from this file (see kconfig.mk)
 osource "$(KCONFIG_GENERATED_DEPENDENCIES)"
 
+rsource "cpu/Kconfig"
```

Comment:
@leandrolanzieri Is it planned that the board will also be selected via Kconfig?

If not, at least the board must be specified at make command line, so the CPU is known. In this case I wonder if it wouldn't be better to include the CPU configuration as follows 
```suggestion
# The CPU may declare new symbols as well
osource "$(RIOTBASE)/cpu/$(CPU)/Kconfig"

```
This would avoid the user having to select the CPU from a list. He would only have one top-level menu entry for the specified CPU, provided the CPU has a `Kconfig' file.

Reply:
> @leandrolanzieri Is it planned that the board will also be selected via Kconfig?

We can move towards this, but not just yet as it has huge implications in the resolution of module dependencies.

> If not, at least the board must be specified at make command line, so the CPU is known.

The configurator gets executed after `$(BOARD)` and `$(CPU)` being defined, always.

> This would avoid the user having to select the CPU from a list. He would only have one top-level menu entry for the specified CPU, provided the CPU has a `Kconfig' file.

I think the way @basilfx is adding the `menuconfig` entry is correct, as it depends on the modules defined for EFM32 CPUs. That means that the user will only see the entry for the CPU that's being used. Right now this is not for selecting a particular CPU but to configure its parameters.

Reply:
Ok, my concern is the usability. To configure the CPU, you have to select the CPU in main, then you have to enable the CPU for configuration first, and then you can enter into the CPU configuration
```
CPU  --->
      [*] Configure EFM32 CPU (NEW) ---> 
                    Features  --->
                [ ] Enable EMLIB assertions (NEW)
```
If there is only one CPU at the same time, why not add `Configure EFM32 CPU (NEW)` directly to the main menu.

I'm just asking because I have the same issue for ESPs and solved it in different way. If you define that there should only be `menu` entries in the `main` menu and no `menuconfig` entries, that would be fine with me and I would do it in the same for the ESPs.

Reply:
We could, in the root Kconfig file, do:
```
rsource "$(RIOTCPU)/*/Kconfig"
```
But if we then start having 'common' Kconfig files then this will add more than one entry in the Top level. Maybe then having a `CPU` menu is better.

Also for that we can just include every CPU using glob patterns:
```
# in /cpu/Kconfig
menu "CPU"

rsource "*/Kconfig"

endmenu

Reply:
I tried the latter, it works. So it is not necessary to include each `cpu/*/Kconfig` file separately.